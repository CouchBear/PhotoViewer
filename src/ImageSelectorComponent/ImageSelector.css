.imageGallery {
    display: flexbox;
    align-self: center;
    width: 100%;
    height: 30%;
    border: 1px solid #ddd;



}

.individualImage {
    border: 1px solid #ddd;
    /* Gray border */
    border-radius: 4px;
    /* Rounded border */
    padding: 5px;
    /* Some padding */
    width: 150px;
    /* Set a small width */
}

.individualImage:hover {
    box-shadow: 0 0 10px 1px rgba(0, 140, 186, 0.5);
}