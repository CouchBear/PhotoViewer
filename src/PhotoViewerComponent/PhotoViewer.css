.displayPhoto {

    width: 100%;
    height: 100%;




}

#div-main {
    width: 100%;
    height: 40%;


}