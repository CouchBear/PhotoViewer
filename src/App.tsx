import { useState } from 'react'
import reactLogo from './assets/react.svg'
import './App.css'
import { PhotoViewer } from './PhotoViewerComponent/PhotoViewer';
import { ImageSelector } from './ImageSelectorComponent/ImageSelector';
import { Nav } from './NavComponent/Nav';

function App() {

  const [selected, setSelected] = useState("https://picsum.photos/id/223/1600/900.jpg");


  return (
    <div>
      <Nav />
      <p className="h1-text">A New World</p>
      <PhotoViewer src={selected} />
      <p className="h1-text">Image Selector</p>
      <ImageSelector selected={selected} setSelected={setSelected} />
    </div>

  );
}

export default App
