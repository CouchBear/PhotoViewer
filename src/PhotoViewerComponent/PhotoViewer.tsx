import React from "react";  // import React (to provide access to TSX)
import "./PhotoViewer.css";
function PhotoViewer(props: string) {    // declare and export new function called 'PhotoViewer'
    return (                
        <div>               
            
          <img src={props.src} className="displayPhoto" /> 

        </div>
    );
}

export {PhotoViewer};