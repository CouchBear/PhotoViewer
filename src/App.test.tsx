import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { PhotoViewer } from './PhotoViewerComponent/PhotoViewer';
import { ImageSelector } from './ImageSelectorComponent/ImageSelector';
import { getImageUrls } from './ImageSelectorComponent/ImageSelector';
import renderer from 'react-test-renderer';
import { create } from 'react-test-renderer';

// test('renders Hello world text', () => {
//   const { getByText } = render(<App />);
//   const textElement = getByText(/Hello world/i);
//   expect(textElement).toBeInTheDocument();
// });



// check that this is the first image
// https://picsum.photos/id/223/1600/900.jpg

test('src contains correct value', () => {
  render(<PhotoViewer src="https://picsum.photos/id/223/1600/900.jpg" />)
  const testImage = document.querySelector("img") as HTMLImageElement;
  expect(testImage.src).toContain("https://picsum.photos/id/223/1600/900.jpg");
});

test('src contains correct image value', () => {

  const imageUrls = getImageUrls();
  expect(imageUrls[0]).toContain("https://picsum.photos/id/600/1600/900.jpg");
})

test('getImagesUrls function does not include broken image', () => {

  const imageUrls = getImageUrls();
  expect(imageUrls).not.toContain("https://picsum.photos/id/601/1600/900.jpg");

})


describe("My Component", () => {
  it("Should match snapshot without src prop", async () => {
    const tree = create(<PhotoViewer src="https://picsum.photos/id/223/1600/900.jpg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
