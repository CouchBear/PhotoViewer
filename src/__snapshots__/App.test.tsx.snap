// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Should match snapshot without src prop 1`] = `
<div
  id="div-main"
>
  <img
    className="displayPhoto"
    src="https://picsum.photos/id/223/1600/900.jpg"
  />
</div>
`;
